lib/vector-core/src/event/mod.rs

```diff
@@ -442,3 +442,16 @@ impl MaybeAsLogMut for Event {
         }
     }
 }
+
+pub trait MaybeAsLog {
+    fn maybe_as_log(&self) -> Option<&LogEvent>;
+}
+
+impl MaybeAsLog for Event {
+    fn maybe_as_log(&self) -> Option<&LogEvent> {
```

Comment:
Yes, this function is what I was describing, but until we actually need to name the trait (ie a generic bound or something), it can just go into the base `impl Event` block and not a new trait.